### PR TITLE
DATAUP-532: Using batch job data for tests

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

Main change in this ticket is adding in some batch job data and updating functions to take account of retried jobs. See inline comments for details.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-532
- [x] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions
- [x] Changes available by spinning up a local narrative and navigating to _X_ to see _Y_

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (JavaScript) I have run Prettier and ESLint on changed code manually or with a git precommit hook

### CODE REVIEW

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 1]
```diff
@@ -619,6 +738,12 @@ define(['common/errorDisplay', 'common/format', 'common/html', 'common/props', '
                 })
                 .join(', ');
 
+        if (statuses.retried) {
+            summaryHtml += ` (${
+                statuses.retried > 1 ? statuses.retried + ' jobs' : '1 job'
+            } retried)`;
+        }
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:745]: The status now reports the most current jobs, and adds in brackets the number of jobs that have been retried (not the number of retries).

File: kbase-extension/static/kbase/js/util/jobLogViewer.js [hunk 2]
```diff
@@ -1249,7 +1249,10 @@ define([
             ui.setContent('log-container', '');
             awaitingLog = false;
             // slightly hacky way to get the appropriate job status lines
-            renderJobState({ job_state: 'does_not_exist' });
+            renderJobState({
+                job_id: '',
+                status: 'does_not_exist',
+            });
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/util/jobLogViewer.js:1255]: remove slightly nasty hack, replace with slightly less nasty hack

File: test/unit/spec/common/jobCommChannel-spec.js [hunk 3]
```diff
@@ -140,7 +140,7 @@ define(['common/jobCommChannel', 'base/js/namespace', 'common/runtime', 'testUti
             },
             {
                 channel: 'request-job-updates-start',
-                message: { jobId: 'someJob', parentJobId: 'someParent' },
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobCommChannel-spec.js:143]: don't know how this snuck in!

File: test/unit/spec/common/jobs-Spec.js [hunk 4]
```diff
@@ -141,17 +145,6 @@ define(['common/jobs', '/test/data/jobsData'], (Jobs, JobsData) => {
             });
         });
 
-        it('should create an appropriate string if the job does not exist', () => {
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobs-Spec.js:144]: removed this hacky stuff from the Jobs implementation => remove this test

File: test/unit/spec/common/jobs-Spec.js [hunk 5]
```diff
@@ -281,91 +274,166 @@ define(['common/jobs', '/test/data/jobsData'], (Jobs, JobsData) => {
         });
     });
 
+    function generateJobs(jobSpec) {
```
Comment (ialarmedalien) [inline test/unit/spec/common/jobs-Spec.js:277]: I was too lazy to redo the input data nicely => wrote a quick function to do it instead!

File: kbase-extension/static/kbase/js/common/jobs.js [hunk 6]
```diff
@@ -407,11 +424,6 @@ define(['common/errorDisplay', 'common/format', 'common/html', 'common/props', '
 
     function createJobStatusLines(jobState, includeHistory = false) {
         if (!isValidJobStateObject(jobState)) {
-            // check whether EE2 sent { job_state: 'does_not_exist' }
-            // TODO: coerce the { job_state: 'does_not_exist' } into a standard format
-            if (jobState && jobState.job_state && jobState.job_state === 'does_not_exist') {
-                return jobNotFound;
-            }
```
Comment (ialarmedalien) [inline kbase-extension/static/kbase/js/common/jobs.js:414]: Get rid of this hack as it is no longer required -- `job_state: "does_not_exist"` should not come from the backend and will not be generated in the frontend.

File: kbase-extension/static/kbase/js/common/props.js [hunk 7]
```diff
@@ -289,23 +289,27 @@ define([], () => {
...
-            popItem: popItem,
-            reset: reset,
-            getRawObject: function () {
-                return obj;
-            },
-            getLastRawObject: function () {
-                return lastObj;
-            },
-            getHistoryCount: getHistoryCount,
+            setItem,
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/props.js:301]: So tidy!

File: kbase-extension/static/kbase/js/common/props.js [hunk 8]
```diff
@@ -289,23 +289,27 @@ define([], () => {
...
+            getItem,
+            copyItem,
+            incrItem,
+            deleteItem,
+            pushItem,
+            popItem,
+            reset,
+            getRawObject,
+            getLastRawObject,
+            getHistoryCount,
```
Comment (briehl) [inline kbase-extension/static/kbase/js/common/props.js:312]: So tidy!

File: kbase-extension/static/kbase/js/util/jobLogViewer.js (hunk 2 above)
Comment (briehl) [reply kbase-extension/static/kbase/js/util/jobLogViewer.js:1255]: 😅

File: test/data/jobsData.js [hunk 9]
```diff
@@ -363,11 +369,161 @@ define(['common/format'], (format) => {
...
+     *  - retry 2: 'job-estimating' (most recent retry)
+     *
+     * Extra metadata for batch jobs:
+     * meta.currentJob: true/false -- this is the most recent job (including retries)
+     * meta.originalJob: true/false -- this is one of the original jobs in the batch
+     */
+
+    function createBatchJob() {
+        // extra metadata:
+        //
```
Comment (briehl) [inline test/data/jobsData.js:432]: I feel like there's some added suspense here! (or maybe that line isn't needed, as the info is above).

**Review by briehl — APPROVED** [review]
LGTM

File: test/data/jobsData.js (hunk 9 above)
Comment (ialarmedalien) [reply test/data/jobsData.js:432]: hah! Those invisible characters catch people out every time!

**Pull request merged**